### PR TITLE
Feature 8 expose user endpoint

### PR DESCRIPTION
**Pull request by Alekseeybg**

### CODE REVIEW

File: src/main/java/jewellery/inventory/service/UserService.java [hunk 1]
```diff
@@ -31,23 +29,18 @@ public UserResponse getUser(UUID id) {
...
+    validateUserDetails(userToCreate);
+    return UserMapper.INSTANCE.toUserResponse(userRepository.save(userToCreate));
   }
 
   public UserResponse updateUser(UserRequest userRequest, UUID id) {
-    User existingUser =
-        userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(id));
+    if (!userRepository.existsById(id)) {
+      throw new UserNotFoundException(id);
+    }
```
Comment (VladoKat) [inline src/main/java/jewellery/inventory/service/UserService.java:40]: Why did you need to change this? I am curious

**Review by VladoKat — COMMENTED** [review]
I'm happy to approve. Just curious about the change with existsById instead of findById

File: src/main/java/jewellery/inventory/service/UserService.java (hunk 1 above)
Comment (Alekseeybg) [reply src/main/java/jewellery/inventory/service/UserService.java:40]: It is supposed to be more performant. With this we are just checking if the user exists while with the old version we checked and had the whole user 'extracted' which was unnecessary

**Review by VladoKat — APPROVED** [review]
Looks good
